public/js/map/overlays/PlayerOverlay.js

```diff
@@ -83,6 +82,53 @@ export default L.LayerGroup.extend({
     });
   },
 
+  getSkin: function(player) {
+    if (!player.skin || player.skin === "" || player.skin === "character.png") return defaultSkin;
+
+    let skin = `api/skins/${player.skin}`;
+
+    if (playerSkins[skin]) return playerSkins[skin];
+
+    // no cached skin, we need to build the image
+    let img = new Image();
+    img.onload = function() {
```

Comment:
The skins will load asyncronously, and then be added to the `playerSkins` map/cache, so after a subsequent update they'll rerender.